### DISCUSSION TITLE
Missing Access-Control-Allow-Origin header with CORS when Origin is not allowed.

### DISCUSSION THREAD

**Discussion by JCHacking** [fastapi/fastapi#10980] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.middleware.cors import CORSMiddleware

app = FastAPI()

app.add_middleware(
    CORSMiddleware,
    allow_origins=["http://test.com"],
    allow_methods=["GET"],
)
```

## Steps to reproduce
* Make request OPTIONS http://localhost:8000 with headers Origin="http://example.com" and Access-Control-Request-Method="GET"
* See response headers don't have Access-Control-Allow-Origin

## Question
Is this expected, or is it a strange behavior? 

### Operating System

Windows

### FastAPI Version

0.109.0

### Pydantic Version

2.5.3

### Python Version

3.12.0

**Comment by hasansezertasan** [comment, marked as answer]
It's not about FastAPI, it's about Starlette 🙏

**Reply by JCHacking** [reply]
All right, so I'm asking there about this topic. 
https://github.com/encode/starlette/discussions/2424
Thank you!
